React/Modules/RCTNativeAnimationManager.m

```diff
@@ -0,0 +1,236 @@
+  
+  RCTAnimationDriverNode *animationDriver =
+    [[RCTAnimationDriverNode alloc] initWithId:animationId
+                                         delay:delay
+                                       toValue:toValue
+                                        frames:frames
+                                       forNode:valueNode
+                                      callBack:callBack];
+  [_activeAnimations addObject:animationDriver];
+  [_animationDrivers setObject:animationDriver forKey:animationId];
+  [animationDriver startAnimation];
+  [self startAnimation];
+}
+
+- (void)stopAnimation:(nonnull NSNumber *)animationId {
+  RCTAnimationDriverNode *driver = [_animationDrivers objectForKey:animationId];
+  if (driver) {
+    [driver removeAnimation];
+    [_animationDrivers removeObjectForKey:animationId];
+    [_activeAnimations removeObject:driver];
+    [_finishedAnimations removeObject:driver];
+  }
+}
+
+- (void)setAnimatedNodeValue:(nonnull NSNumber *)nodeTag
+                       value:(nonnull NSNumber *)value {
+  RCTAnimatedNode *node = [self nodeForTag:nodeTag];
+  if (!node ||
+      ![node isKindOfClass:[RCTValueAnimatedNode class]]) {
+    NSLog(@"Not a vlue node. Exception");
+    return;
+  }
+  RCTValueAnimatedNode *valueNode = (RCTValueAnimatedNode *)node;
+  valueNode.value = value;
+  [valueNode setNeedsUpdate];
+}
+
+- (void)connectAnimatedNodeToView:(nonnull NSNumber *)nodeTag
+                          viewTag:(nonnull NSNumber *)viewTag {
+  RCTAnimatedNode *node = [self nodeForTag:nodeTag];
+  if (viewTag && node && [node isKindOfClass:[RCTPropsAnimatedNode class]]) {
+    [(RCTPropsAnimatedNode *)node connectToView:viewTag];
+  }
+}
+
+- (void)disconnectAnimatedNodeFromView:(nonnull NSNumber *)nodeTag
+                               viewTag:(nonnull NSNumber *)viewTag {
+  RCTAnimatedNode *node = [self nodeForTag:nodeTag];
+  if (viewTag && node && [node isKindOfClass:[RCTPropsAnimatedNode class]]) {
+    [(RCTPropsAnimatedNode *)node disconnectFromView:viewTag];
+  }
+}
+
+- (void)dropAnimatedNode:(nonnull NSNumber *)tag {
+  NSLog(@"Dropping animated node %@", tag);
+  RCTAnimatedNode *node = [self nodeForTag:tag];
+  if (node) {
+    [node dettachNode];
+    [_animationNodes removeObjectForKey:tag];
+    [_updatedValueNodes removeObject:node];
+    [_propAnimationNodes removeObject:node];
+  }
+}
+
+- (RCTAnimatedNode *)nodeForTag:(NSNumber *)tag {
+  return [_animationNodes objectForKey:tag];
+}
+
+#pragma mark -- Animation Loop
+
+- (void)startAnimation {
+  if (!displayLink_ && _activeAnimations.count > 0) {
+    displayLink_ = [CADisplayLink displayLinkWithTarget:self selector:@selector(updateAnimations)];
+    [displayLink_ addToRunLoop:[NSRunLoop mainRunLoop] forMode:NSRunLoopCommonModes];
+  }
+}
+
+- (void)updateAnimations {
+  // Step Current active animations
+  // This also recursively marks children nodes as needing update
+  [_activeAnimations makeObjectsPerformSelector:@selector(stepAnimation)];
```

Comment:
See note above about this makeObjectsPerformSelector being expensive. It's fine normally, but probably best not to use it inside such a performance critical part of the code.